### PR TITLE
Fix invalid constraints merging

### PR DESCRIPTION
**Pull request by lherman-cs**
the constraints return by `selectBestDriver` should not be set to what the user specified, but instead the best property returned by the driver.

### CODE REVIEW

File: mediadevices.go [hunk 1]
```diff
@@ -218,7 +218,6 @@ func selectBestDriver(filter driver.FilterFn, constraints MediaTrackConstraints)
 	}
 
 	constraints.selectedMedia = bestProp
-	constraints.selectedMedia.Merge(constraints.MediaConstraints)
```
Comment (at-wat) [inline mediadevices.go:221]: I think it is needed for the parameters which can be given from continuous range of the value.
Currently the main usage is FrameRate of screen capture driver.

Comment (lherman-cs) [reply mediadevices.go:221]: Yep, that makes sense.

**Comment by lherman-cs** [comment]
@at-wat just updated. PTAL. It's still not perfect and it feels off, but I think it's good for now to at least fix the issue. 

We should definitely consider to refractor the code base to use `prop.MediaConstraints` vs `prop.Media` properly. I've created https://github.com/pion/mediadevices/issues/178 to track on this.

File: mediadevices.go [hunk 2]
```diff
@@ -217,8 +217,8 @@ func selectBestDriver(filter driver.FilterFn, constraints MediaTrackConstraints)
 		return nil, MediaTrackConstraints{}, errNotFound
 	}
 
-	constraints.selectedMedia = bestProp
-	constraints.selectedMedia.Merge(constraints.MediaConstraints)
+	constraints.selectedMedia.MergeConstraints(constraints.MediaConstraints)
```
Comment (at-wat) [inline mediadevices.go:220]: It would be safer to clear `constraints.selectedMedia` by `prop.Media{}` before merge.

Comment (lherman-cs) [reply mediadevices.go:220]: Yep, that makes sense.

**Pull request merged**